src/mqt/yaqs/core/data_structures/networks.py

```
@@ -745,39 +745,69 @@ def expect(self, observable: Observable) -> np.float64:
-            projected_tensor = oe.contract("a, acd->cd", selected_state, tensor)
+            # Multiply state: project the rotated tensor onto the selected state.
+            oe.contract("a, acd->cd", selected_state, rotated_tensor)
+
             # Propagate the measurement to the next site.
             if site != self.length - 1:
+                # We need to use the original (unrotated) tensor projected onto the rotated basis state
+                # The rotated basis state in the original basis is: rotation^H @ selected_state
+                original_basis_selection = oe.contract("ab, a->b", np.conj(rotation), selected_state)
+                projected_tensor = oe.contract("b, bcd->cd", original_basis_selection, tensor)
+
                 temp_state.tensors[site + 1] = (  # noqa: B909
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `B909`)

Remove unused `noqa` directive

(RUF100)